integration-cli/docker_api_exec_test.go

```diff
@@ -198,3 +219,29 @@ func inspectExec(c *check.C, id string, out interface{}) {
 	err = json.NewDecoder(body).Decode(out)
 	c.Assert(err, checker.IsNil)
 }
+
+func waitForExec(c *check.C, id string) {
+	timeout := time.After(60 * time.Second)
+	var execJSON struct{ Running bool }
+	for {
+		select {
+		case <-timeout:
+			c.Fatal("timeout waiting for exec to start")
+		default:
+		}
+
+		inspectExec(c, id, &execJSON)
+		if !execJSON.Running {
+			break
+		}
+	}
+}
+
+func inspectContainer(c *check.C, id string, out interface{}) {
```

Comment:
you can use `inspectFieldJSON` (https://github.com/docker/docker/blob/master/integration-cli/docker_utils_test.go#L554) instead.

The fix LGTM. 

It's better to squash your commits :)